dotnet/src/SemanticKernel.Core/Extensions/KernelSemanticFunctionExtensions.cs

```diff
@@ -285,4 +313,91 @@ public static class KernelSemanticFunctionExtensions
             kernel.LoggerFactory
         );
     }
+
+    private const string BasicTemplateFactoryAssemblyName = "Microsoft.SemanticKernel.TemplateEngine.Basic";
+    private const string BasicTemplateFactoryTypeName = "BasicPromptTemplateFactory";
+    private static bool s_promptTemplateFactoryInitialized = false;
+    private static Type? s_promptTemplateFactoryType = null;
+
+    /// <summary>
+    /// Create a default prompt template factory.
+    ///
+    /// This is a temporary solution to avoid breaking existing clients.
+    /// There will be a separate task to add support for registering instances of IPromptTemplateEngine and obsoleting the current approach.
+    ///
+    /// </summary>
+    /// <param name="kernel">Kernel instance</param>
+    /// <returns>Instance of <see cref="IPromptTemplateEngine"/>.</returns>
+    private static IPromptTemplateFactory CreateDefaultPromptTemplateFactory(IKernel kernel)
+    {
+#pragma warning disable CS0618 // Type or member is obsolete
+        if (kernel.PromptTemplateEngine is not null)
+        {
+            return new PromptTemplateFactory(kernel.PromptTemplateEngine);
+        }
+#pragma warning restore CS0618 // Type or member is obsolete
+
+        if (!s_promptTemplateFactoryInitialized)
+        {
+            s_promptTemplateFactoryType = GetPromptTemplateFactoryType();
```

Comment:
All this reflection suggests a problem in layering.  If this core assembly needs access to the default template factory, shouldn't the default template factory be in this assembly?

Reply:
I think you may be right. I have a follow up task to remove this code. Roger has suggested an approach but if that doesn't work I may move the code back to the core package.

Tracking task: https://github.com/microsoft/semantic-kernel/issues/3353